src/core/textures/Texture.js

```diff
@@ -216,15 +224,17 @@ export default class Texture extends EventEmitter
     {
         if (this.baseTexture)
         {
-            if (destroyBase)
+            // remove BaseTexture stored under the texture name if destroyBase is true or
+            // is our base texture (in which case this is the event callback for 'destroy')
+            if ((destroyBase === true || destroyBase === this.baseTexture)
```

Comment:
The `destroy` event passes the base texture that was destroyed as a param, so detecting that means detecting when this is an event callback.